executor/runtime/docker/docker_image.go

```diff
@@ -0,0 +1,70 @@
+package docker
+
+import (
+	"strings"
+
+	"github.com/docker/distribution/reference"
+	"github.com/docker/docker/api/types"
+)
+
+// cleanContainerVersion tries to provide the most human-friendly version string we can get for
+// a particular image, given a version inspect.
+// It goes from most human-friendly to least (sha)
+func cleanContainerVersion(i *types.ImageInspect) string {
+	tag := getImageTagFromInspect(i)
+	if tag != "" {
+		return tag
+	}
+	buildLabel := getBuildLabelFromInspect(i)
+	if buildLabel != "" {
+		return buildLabel
+	}
+	return getImageShaFromInspect(i)
+}
+
+func getImageTagFromInspect(i *types.ImageInspect) string {
+	if len(i.RepoTags) < 1 {
+		return ""
+	}
+	return i.RepoTags[0]
+}
+
+func getBuildLabelFromInspect(i *types.ImageInspect) string {
+	if i.Config == nil {
+		return ""
+	}
+	if i.Config.Labels == nil {
+		return ""
+	}
+	nameLabel, ok := i.Config.Labels["image-name"]
+	if ok {
+		versionLabel, ok := i.Config.Labels["image-version"]
+		if ok {
+			return "image:" + fullImageNameToShortName(nameLabel) + " build:" + versionLabel
+		}
+	}
+	return ""
+}
+
+func getImageShaFromInspect(i *types.ImageInspect) string {
+	if len(i.RepoDigests) < 1 {
+		return ""
+	}
+	name := fullImageNameToShortName(i.RepoDigests[0])
+	splitDigest := strings.Split(i.RepoDigests[0], "@")
+	if len(splitDigest) == 0 {
+		return i.RepoDigests[0]
+	}
+	shortDigest := splitDigest[1]
+	return "image:" + name + " digest:" + shortDigest
+}
+
+func fullImageNameToShortName(i string) string {
+	ref, err := reference.ParseNamed(i)
+	if err != nil {
+		return i
+	}
+	fullURL := ref.Name()
+	splitURL := strings.Split(fullURL, "/")
+	return strings.Join(splitURL[1:], "/")
```

Comment:
Not sure this is safe unless there is some guarantee that image names in these contexts will always include a registry (i.e. `busybox` is a vaild reference).

Reply:
This can be simplified to:

```go
return reference.Path(ref)
```